diff --git a/src/utilities/buildable-libs-utils.js b/src/utilities/buildable-libs-utils.js
index 62a8a7b64c58b53254bf30028c7096500e70bf76..7914b1fa6df6a9c95511585c15f2db4d33861ee9 100644
--- a/src/utilities/buildable-libs-utils.js
+++ b/src/utilities/buildable-libs-utils.js
@@ -196,7 +196,7 @@ function updatePaths(dependencies, paths) {
         // If there are outputs
         if (dep.outputs && dep.outputs.length > 0) {
             // Directly map the dependency name to the output paths (dist/packages/..., etc.)
-            paths[dep.name] = dep.outputs;
+            paths[dep.name] = dep.outputs.map(out => out + '/src/index.d.mts');
             // check for secondary entrypoints
             // For each registered path
             for (const path of pathsKeys) {
